submissions/Harnytskyi/friends-app/js/script.js

```diff
@@ -0,0 +1,151 @@
+let friends = [];
+
+let selectedFriends = [...friends];
+function filter() {
+    if (gender.male == true && gender.female == true)
+        selectedFriends = friends;
+    else if (gender.male == true && gender.female == false)
+        selectedFriends = friends.filter(item => item.gender == "male")
+    else if (gender.male == false && gender.female == true)
+        selectedFriends = friends.filter(item => item.gender == "female")
+}
+function displayFriends() {
+    GRID.innerHTML = '';
+    selectedFriends.forEach(item => {
+        GRID.innerHTML +=  `<div id="${item.name}" class="card">
+            <img class="photo" src="${item.photo}">
+            <p class="name">${item.name}</p>
+            <p class="age">Age: ${item.age}</p>
+            <a class="phone" href="${item.phone}">${item.phone}</a>
+            <div class="email">
+                <a href="${item.email}">
+                    <button class="button-message">SEND MESSAGE</button>
+                </a>
+            </div>
+        <div>`
+    });
+    search();
+}
+let gender = {
+    male: true,
+    female: true
+}
+
+function changeFilterStatus(checkbox, sex) {
+    if (checkbox.checked) {
+        gender[sex] = true;
+    } else {
+        if (sex == "male") {
+            if (gender.female == false) {
+                gender.female = true;
+                CHECKBOX_WOMAN.click();
+            }
+        }
+        else if (sex == "female") {
+            if (gender.male == false) {
+                gender.male = true;
+                CHECKBOX_MAN.click();
+            }
+        }
+        gender[sex] = false;
+    }
+    filter();
+    displayFriends();
+}
+
+function search() {
+    let searchRequest = SEARCH_INPUT.value.toUpperCase();
+    selectedFriends.forEach(item => {
+        const FRIEND = document.getElementById(item.name)
+        FRIEND.classList.remove('hidden');
+        if (item.name.toUpperCase().indexOf(searchRequest) == -1) {
+            FRIEND.classList.add('hidden');
+        }
+    })
+}
+
+function sortFriends(criterion, order) {
+    friends.sort((a, b) => {
+        if (order == "descending") {
+            let c = a;
+            a = b;
+            b = c;
+        }
+        if (a[criterion] > b[criterion]) return 1;
+        if (a[criterion] == b[criterion]) return 0;
+        if (a[criterion] < b[criterion]) return -1;
+    })
+    filter();
+    displayFriends();
+}
+
+CHECKBOX_MAN.addEventListener('change', function () {
+    changeFilterStatus(CHECKBOX_MAN, "male");
+});
+CHECKBOX_WOMAN.addEventListener('change', function () {
+    changeFilterStatus(CHECKBOX_WOMAN, "female");
+});
+SORT_AZ.addEventListener('click', function () {
+    sortFriends("name");
+})
+SORT_ZA.addEventListener('click', function () {
+    sortFriends("name", "descending");
+})
+SORT_09.addEventListener('click', function () {
+    sortFriends("age");
+})
+
+SORT_90.addEventListener('click', function () {
+    sortFriends("age", "descending");
```

Comment:
Event delegation

Reply:
What is wrong? Can't use Event delegation?

Reply:
united two event listeners into a common one `FILTER_SORT`